cmd/mattermost/commands/config.go

```diff
@@ -343,6 +352,112 @@ func UpdateMap(configMap map[string]interface{}, configSettings []string, newVal
 	}
 }
 
+func configResetCmdF(command *cobra.Command, args []string) error {
+	app, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+
+	defer app.Shutdown()
+
+	oldConfig := app.Config()
+	newConfig := app.Config()
+	defaultConfig := &model.Config{}
+	defaultConfig.SetDefaults()
+
+	if len(args) == 0 {
+		var confirmResetAll string
+		CommandPrettyPrintln("Are you sure you want to reset all the configuration settings?(YES/NO): ")
+		fmt.Scanln(&confirmResetAll)
+		if confirmResetAll == "YES" {
+			oldConfigMap := configToMap(*oldConfig)
+			for key := range oldConfigMap {
+				f := resetConfigValue(key, defaultConfig, oldConfig, newConfig)
+				app.UpdateConfig(f)
+				if err := newConfig.IsValid(); err != nil {
+					return err
+				}
+				if changed := config.FixInvalidLocales(newConfig); changed {
+					return errors.New("Invalid locale configuration")
+				}
+			}
+		}
+	}
+
+	for _, arg := range args {
+		fmt.Println(arg)
+		f := resetConfigValue(arg, defaultConfig, oldConfig, newConfig)
+		app.UpdateConfig(f)
+		if err := newConfig.IsValid(); err != nil {
+			return err
+		}
+		if changed := config.FixInvalidLocales(newConfig); changed {
+			return errors.New("Invalid locale configuration")
+		}
+	}
+
+	app.PersistConfig()
+	return nil
+}
+
+func resetConfigValue(configSetting string, defaultConfig, oldConfig, newConfig *model.Config) func(*model.Config) {
+	return func(update *model.Config) {
+		oldConfigMap := configToMap(*oldConfig)
+		defaultConfigMap := configToMap(*defaultConfig)
+
+		err := resetMap(oldConfigMap, defaultConfigMap, strings.Split(configSetting, "."))
+		if err != nil {
+			fmt.Printf("%s\n", err)
+			os.Exit(1)
+		}
+
+		bs, err := json.Marshal(oldConfigMap)
+		if err != nil {
+			fmt.Printf("Error while marshalling map to json %s\n", err)
+			os.Exit(1)
+		}
+
+		err = json.Unmarshal(bs, newConfig)
```

Comment:
My proposed change affect this too, It should be something like:
```suggestion
		err = json.Unmarshal(bs, update)
```